### ISSUE TITLE
Helm Repository Resource Persistance Issue

### ISSUE THREAD

**Issue by Jeeppler** [hashicorp/terraform-provider-helm#213]
Helm repositories are not added to `repositories.yaml` after initializing Helm with `helm init --client-only`. 

### Terraform Version

Terraform v0.11.11
+ provider.aws v1.58.0
+ provider.helm v0.8.0

### Affected Resource(s)

- helm_repository

### Terraform Configuration Files

```hcl
resource "helm_repository" "incubator" {
  name = "incubator"
  url  = "https://kubernetes-charts-incubator.storage.googleapis.com"
}
```

### Expected Behavior

The `helm_repository` resource should be added to the local `~/.helm/repository/repositories.yaml` during Terraform plan if the repository is not in the list, unless the `~/.helm/repository/repositories.yaml` does not exist.

The `helm_repository` should behave like the [`local_file`](https://www.terraform.io/docs/providers/local/r/file.html) resource of the `Local` provider.

The Helm provider should not expect that files on the local computer are in a certain state. Instead it  should rely on the Terraform state file.

\---

Use case in which this becomes an issue:

Let's assume Alice and Bob are part of a team and their Terraform scripts are stored in a Git repository. The Terraform state is stored remotely. Everybody, installs Helm and Terraform. Team member Alice sets up the Helm charts incubator repository using `helm_repository`  and installs several charts from the incubator repository. At this point Alice has the incubator repository in her `~/.helm/repository/repositories.yaml` file.
Something did not work and Bob has to remove the incubator repositories. Bob clones the Git repositories to his local computer and run `terraform destroy`. However, since Bob is working on a different computer Bob does not have the incubator repository in his local `~/.helm/repository/repositories.yaml` file. Currently, that would lead to an error. However, the expected behaviour would be that Bob can run the Terraform script and the Terraform Helm provider adds the `helm_repository` resource to Bob local environment.

### Actual Behavior

The `helm_repository` resource adds, during the creation of the resource, the repository URL to the local Helm repository file `~/.helm/repository/repositories.yaml`. However, if you delete the `~/.helm/` folder and you initialize Helm again with `helm init --client-only` as well as run `terraform plan` again you will get the following error message:

```
Error: Error refreshing state: 1 error(s) occurred:

* helm_repository.incubator: 1 error(s) occurred:

* helm_repository.incubator: helm_repository.incubator: repository not found
```

The reason is that `helm init --client-only` will not add the `helm_repository` resource again to the repository. 

`helm init --client-only` does the following:

```
$ helm init --client-only
Creating /home/user/.helm 
Creating /home/user/.helm/repository 
Creating /home/user/.helm/repository/cache 
Creating /home/user/.helm/repository/local 
Creating /home/user/.helm/plugins 
Creating /home/user/.helm/starters 
Creating /home/user/.helm/cache/archive 
Creating /home/user/.helm/repository/repositories.yaml 
Adding stable repo with URL: https://kubernetes-charts.storage.googleapis.com 
Adding local repo with URL: http://127.0.0.1:8879/charts 
```

As you can see `helm init --client-only` does not add an incubator or any other custom repository to the `repositories.yaml` file, but it adds the stable repo to the file, therefore the stable repository will not have the issue.

In my opinion the `helm_repository` resource should be added to the `respositories.yaml` file, if the repository is not in the list.

### Steps to Reproduce

Prerequisites:

- Use the `incubator` repository to install any resource:

```hcl
resource "helm_repository" "incubator" {
  name = "incubator"
  url  = "https://kubernetes-charts-incubator.storage.googleapis.com"
}

# For example install the 
# [Kubeless chart](https://github.com/helm/charts/tree/master/incubator/kubeless)
resource "helm_release" "kubeless" {
  name       = "kubeless"
  repository = "${helm_repository.incubator.metadata.0.name}"
  chart      = "kubeless"
  version    = "2.0.1"
  namespace  = "kubeless"

  set {
    name  = "rbac.create"
    value = "true"
  }
}
```

Steps:

1. `terraform plan`
2. `terraform apply`
3. Delete the local Helm folder `~/.helm/` (e.g. `rm -rf ~/.helm/`).
4. Do a `helm init --client-only`
5. `terraform plan`

### Workaround

Add the repository manually, if it does not exist via Helm CLI. For example, for the [`incubator` repository](https://github.com/helm/charts/tree/master/incubator): 

```
helm repo add incubator https://kubernetes-charts-incubator.storage.googleapis.com
```

**Comment by meyskens** [comment]
Closing this as a duplicate of #146. This will be fixed in #221

**Comment by Jeeppler** [comment]
@meyskens thanks for the references.
